### PR TITLE
check if eval dataset is dict

### PR DESCRIPTION
**Pull request by hwijeen**
# What does this PR do?
Simply checks if `eval_dataset` is a dict and if it is, run a sequential evaluation on each evaluation dataset.

Fixes # (issue)
#24832 

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?
@sgugger 
Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: examples/pytorch/summarization/run_summarization.py [hunk 1]
```diff
@@ -692,7 +692,16 @@ def compute_metrics(eval_preds):
...
     if training_args.do_eval:
         logger.info("*** Evaluate ***")
-        metrics = trainer.evaluate(metric_key_prefix="eval")
+        if isinstance(eval_dataset, dict):
+            metrics = {}
+            for eval_ds_name, eval_ds in eval_dataset.items():
+                dataset_metrics = trainer.evaluate(
+                    eval_dataset=eval_ds,
+                    metric_key_prefix=f"eval_{eval_ds_name}",
+                )
```
Comment (sgugger) [inline examples/pytorch/summarization/run_summarization.py:701]: This should fit in one line here.

**Review by sgugger — APPROVED** [review]
Thanks a lot for your PR! Left one small comment.

File: examples/pytorch/summarization/run_summarization.py (hunk 1 above)
Comment (hwijeen) [reply examples/pytorch/summarization/run_summarization.py]: Thanks! Just reformatted. What is the max line length Huggingface prefers?

Comment (hwijeen) [reply examples/pytorch/summarization/run_summarization.py]: CircleCI throws this error message. I don't see how my commit is relevant to this?

```
FAILED examples/pytorch/test_pytorch_examples.py::ExamplesTests::test_generation - OSError: There was a specific connection error when trying to load sshleifer/tiny-gpt2:
421 Client Error: Misdirected Request for url: https://huggingface.co/sshleifer/tiny-gpt2/resolve/main/model.safetensors.index.json
FAILED examples/pytorch/test_pytorch_examples.py::ExamplesTests::test_run_audio_classification - OSError: There was a specific connection error when trying to load hf-internal-testing/tiny-random-wav2vec2:
421 Client Error: Misdirected Request for url: https://huggingface.co/hf-internal-testing/tiny-random-wav2vec2/resolve/main/preprocessor_config.json
FAILED examples/pytorch/test_pytorch_examples.py::ExamplesTests::test_run_clm - OSError: There was a specific connection error when trying to load distilgpt2:
421 Client Error: Misdirected Request for url: https://huggingface.co/distilgpt2/resolve/main/config.json
FAILED examples/pytorch/test_pytorch_examples.py::ExamplesTests::test_run_clm_config_overrides - OSError: There was a specific connection error when trying to load gpt2:
421 Client Error: Misdirected Request for url: https://huggingface.co/gpt2/resolve/main/config.json
FAILED examples/pytorch/test_pytorch_examples.py::ExamplesTests::test_run_image_classification - OSError: There was a specific connection error when trying to load google/vit-base-patch16-224-in21k:
421 Client Error: Misdirected Request for url: https://huggingface.co/google/vit-base-patch16-224-in21k/resolve/main/config.json
FAILED examples/pytorch/test_pytorch_examples.py::ExamplesTests::test_run_mlm - OSError: There was a specific connection error when trying to load distilroberta-base:
421 Client Error: Misdirected Request for url: https://huggingface.co/distilroberta-base/resolve/main/model.safetensors
FAILED examples/pytorch/test_pytorch_examples.py::ExamplesTests::test_run_ner - OSError: There was a specific connection error when trying to load bert-base-uncased:
421 Client Error: Misdirected Request for url: https://huggingface.co/bert-base-uncased/resolve/main/config.json
FAILED examples/pytorch/test_pytorch_examples.py::ExamplesTests::test_run_semantic_segmentation - huggingface_hub.utils._errors.HfHubHTTPError: 421 Client Error: Misdirected Request for url: https://huggingface.co/datasets/huggingface/semantic-segmentation-test-sample/resolve/main/id2label.json
FAILED examples/pytorch/test_pytorch_examples.py::ExamplesTests::test_run_speech_recognition_ctc -   File "/home/circleci/.cache/huggingface/modules/evaluate_modules/metrics/evaluate-metric--wer/1f0994d772b2a524d0447b9618e48c30fde5dc5b736ad5bd0cc76e6ccb600d26/wer.py", line 1
    <!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
    ^
SyntaxError: invalid syntax
```

**Review by sgugger — APPROVED** [review]

File: examples/pytorch/summarization/run_summarization.py (hunk 1 above)
Comment (sgugger) [reply examples/pytorch/summarization/run_summarization.py]: It was a flaky error, re-running the tests show everything is fine.

**Pull request merged**